cloud/linode/instances.go

```diff
@@ -29,176 +29,89 @@ func (e instanceNoIPAddressesError) Error() string {
-	ctx = sentry.SetHubOnContext(ctx)
 	sentry.SetTag(ctx, "provider_id", providerID)
+	sentry.SetTag(ctx, "node_name", node.Name)
 
-	id, err := parseProviderID(providerID)
-	if err != nil {
-		sentry.CaptureError(ctx, err)
-		return nil, err
-	}
-
-	linode, err := linodeByID(ctx, i.client, id)
-	if err != nil {
-		sentry.CaptureError(ctx, err)
-		return nil, err
-	}
-
-	addresses, err := i.nodeAddresses(ctx, linode)
-	if err != nil {
-		sentry.CaptureError(ctx, err)
-		return nil, err
-	}
-
-	return addresses, nil
-}
-
-func (i *instances) nodeAddresses(ctx context.Context, linode *linodego.Instance) ([]v1.NodeAddress, error) {
-	var addresses []v1.NodeAddress
-	addresses = append(addresses, v1.NodeAddress{Type: v1.NodeHostName, Address: linode.Label})
-
-	ips, err := i.client.GetInstanceIPAddresses(ctx, linode.ID)
-	if err != nil {
-		return nil, err
-	}
-
-	if (len(ips.IPv4.Public) == 0) && (len(ips.IPv4.Private) == 0) {
-		return nil, instanceNoIPAddressesError{linode.ID}
-	}
-
-	if len(ips.IPv4.Public) > 0 {
-		addresses = append(addresses, v1.NodeAddress{Type: v1.NodeExternalIP, Address: ips.IPv4.Public[0].Address})
-	}
-	// Allow Nodes to not have an ExternalIP, if this proves problematic this will be reverted
+	if providerID != "" {
+		id, err := parseProviderID(providerID)
+		if err != nil {
+			sentry.CaptureError(ctx, err)
+			return nil, err
+		}
+		sentry.SetTag(ctx, "linode_id", strconv.Itoa(id))
 
-	if len(ips.IPv4.Private) > 0 {
-		addresses = append(addresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: ips.IPv4.Private[0].Address})
+		return linodeByID(ctx, i.client, id)
 	}
-	// Allow Nodes to not have an InternalIP, if this proves problematic this will be reverted
 
-	return addresses, nil
+	return linodeByName(ctx, i.client, nodeName)
 }
 
-func (i *instances) InstanceID(ctx context.Context, nodeName types.NodeName) (string, error) {
+func (i *instances) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
 	ctx = sentry.SetHubOnContext(ctx)
-	sentry.SetTag(ctx, "node_name", string(nodeName))
-
-	linode, err := linodeByName(ctx, i.client, nodeName)
-	if err != nil {
+	if _, err := i.lookupLinode(ctx, node); err != nil {
+		if apiError, ok := err.(*linodego.Error); ok && apiError.Code == http.StatusNotFound {
+			return false, nil
+		}
 		sentry.CaptureError(ctx, err)
-		return "", err
+		return false, err
 	}
-	return strconv.Itoa(linode.ID), nil
-}
 
-func (i *instances) InstanceType(ctx context.Context, nodeName types.NodeName) (string, error) {
-	ctx = sentry.SetHubOnContext(ctx)
-	sentry.SetTag(ctx, "node_name", string(nodeName))
-
-	linode, err := linodeByName(ctx, i.client, nodeName)
-	if err != nil {
-		sentry.CaptureError(ctx, err)
-		return "", err
-	}
-	return linode.Type, nil
+	return true, nil
 }
 
-func (i *instances) InstanceTypeByProviderID(ctx context.Context, providerID string) (string, error) {
+func (i *instances) InstanceShutdown(ctx context.Context, node *v1.Node) (bool, error) {
```

Comment:
Same here can we rename to a true / false question.

Maybe something like this
```suggestion
func (i *instances) IsInstanceShutdown(ctx context.Context, node *v1.Node) (bool, error) {
```

Reply:
Addressed above.